src/wp-includes/rest-api.php

```diff
@@ -1823,3 +1831,112 @@ function rest_default_additional_properties_to_false( $schema ) {
 
 	return $schema;
 }
+
+/**
+ * Gets the REST API route for a post.
+ *
+ * @since 5.5.0
+ *
+ * @param WP_Post $post The post object.
+ * @return string The route path with a leading slash for the given post, or an empty string if there is not a route.
+ */
+function rest_get_route_for_post( $post ) {
+	if ( ! $post instanceof WP_Post ) {
```

Comment:
I'm wondering if we should allow for passing a post object or ID. This is what you can do in 90% of the codebase when passing a "post reference".

```suggestion
	$post = get_post( $post );
	if ( ! $post ) {
```

Thoughts?

Reply:
I think that makes sense, added in c9e7dab.